src/taco-tests-utils/projectHelper.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ *******************************************************
+ *                                                     *
+ *   Copyright (C) Microsoft. All rights reserved.     *
+ *                                                     *
+ *******************************************************
+ */
+
+"use strict";
+
+/* tslint:disable:no-var-requires */
+// var require needed for should module to work correctly
+// Note not import: We don't want to refer to shouldModule, but we need the require to occur since it modifies the prototype of Object.
+var shouldModule: any = require("should");
+/* tslint:enable:no-var-requires */
+
+import child_process = require ("child_process");
+import fs = require ("fs");
+import os = require ("os");
+import path = require ("path");
+import Q = require ("q");
+
+module TacoTestsUtils {
+    interface IKeyValuePair<T> {
+        [key: string]: T;
+    }
+
+    export class ProjectHelper {
+        public static checkPlatformVersions(platformsExpected: IKeyValuePair<string>, projectPath: string): Q.Promise<any> {
+            var platformsInstalled: string[] = ProjectHelper.getInstalledPlatforms(platformsExpected, projectPath);
+            var onWindows: boolean = process.platform === "win32";
+            var deferred: Q.Deferred<any> = Q.defer<any>();
```

Comment:
I think this deferred should be inside the platformsInstalled.map function, otherwise as soon as the first platform finishes everything will be considered done (since there is only one deferred, returned multiple times)